opensrp-pnc/src/main/java/org/smartregister/pnc/utils/PncUtils.java

```diff
@@ -376,4 +378,201 @@ public static void processEvents(@NonNull List<PncEventClient> pncEventClients)
+                    button.setTextColor(ContextCompat.getColor(button.getContext(), R.color.dark_grey));
+                    button.setBackground(ContextCompat.getDrawable(button.getContext(), R.drawable.pnc_btn_done_today));
+                } else if (pncVisitScheduler.getStatus() == VisitStatus.PNC_CLOSE) {
+                    button.setText(R.string.pnc_close);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.pnc_close);
+                }
+            }
+        }
+    }
+
+    public static void addGlobals(String baseEntityId, JSONObject form) {
+
+        String q1 = "SELECT * FROM ec_client WHERE base_entity_id = '" + baseEntityId + "'";
+        String q2 = "SELECT * FROM pnc_registration_details WHERE base_entity_id = '" + baseEntityId + "'";
+        String q3 = "SELECT dob AS baby_dob FROM pnc_baby WHERE mother_base_entity_id = '" + baseEntityId + "'";
+
+        Map<String, String> detailMap = getMergedData(q1, q2, q3);
+
+        try {
+            JSONObject defaultGlobal = new JSONObject();
+
+            for (Map.Entry<String, String> entry : detailMap.entrySet()) {
+                defaultGlobal.put(entry.getKey(), entry.getValue());
+            }
+
+            LocalDate todayDate = LocalDate.now();
+            if (detailMap.get(PncConstants.FormGlobalConstants.DELIVERY_DATE) != null) {
+                LocalDate deliveryDate = LocalDate.parse(detailMap.get(PncConstants.FormGlobalConstants.DELIVERY_DATE), DateTimeFormat.forPattern("dd-MM-yyyy"));
+                int numberOfDays = Days.daysBetween(deliveryDate, todayDate).getDays();
+                defaultGlobal.put(PncConstants.FormGlobalConstants.PNC_VISIT_PERIOD, numberOfDays);
+            }
+
+            if (detailMap.get(PncConstants.FormGlobalConstants.BABY_DOB) != null) {
+                LocalDate babyDob = LocalDate.parse(detailMap.get(PncConstants.FormGlobalConstants.BABY_DOB), DateTimeFormat.forPattern("dd-MM-yyyy"));
+                int numberOfYears = new Period(babyDob, todayDate).getYears();
+                defaultGlobal.put(PncConstants.FormGlobalConstants.BABY_AGE, numberOfYears);
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS, detailMap.get(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS));
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT, detailMap.get(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT));
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS, detailMap.get(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS));
+            }
```

Comment:
Is this necessary?

Reply:
yes, i have added the condition for event type